### PR TITLE
[deploy] Add basedir to files

### PR DESCRIPTION
**Pull request by jaylees14**
So we can run the deployment script from anywhere, make each path used include the base part of the directory too.

Test Plan:

From any directory run `source path/to/spec-golang/deploy.sh`

### CODE REVIEW

File: deploy.sh [hunk 1]
```diff
@@ -12,11 +14,11 @@ minikube start --vm-driver=virtualbox
 echo $GREEN
 
 kubectl create secret docker-registry regcred --docker-server=$REG_URL --docker-username=$REG_USERNAME --docker-password=$REG_PASSWORD --docker-email=$REG_EMAIL
-kubectl create configmap match-db-config --from-file match-db/init.sql -o=yaml
-kubectl create configmap user-db-config --from-file user-db/init.sql -o=yaml
-kubectl create configmap auth-db-config --from-file auth-db/init.sql -o=yaml
+kubectl create configmap match-db-config --from-file $BASEDIR/match-db/init.sql -o=yaml
+kubectl create configmap user-db-config --from-file $BASEDIR/user-db/init.sql -o=yaml
+kubectl create configmap auth-db-config --from-file $BASEDIR/auth-db/init.sql -o=yaml
```
Comment (c-harding) [inline deploy.sh]:
```suggestion
kubectl create configmap match-db-config --from-file "$BASEDIR/match-db/init.sql" -o=yaml
kubectl create configmap user-db-config --from-file "$BASEDIR/user-db/init.sql" -o=yaml
kubectl create configmap auth-db-config --from-file "$BASEDIR/auth-db/init.sql" -o=yaml
```

File: deploy.sh [hunk 2]
```diff
@@ -50,7 +52,7 @@ echo Configuring Kong...
 
 sleep 1
 
-sh kong/configure-kong-k8s.sh
+sh $BASEDIR/kong/configure-kong-k8s.sh
```
Comment (c-harding) [inline deploy.sh]:
```suggestion
sh "$BASEDIR/kong/configure-kong-k8s.sh"
```

**Review by willjvsmith — APPROVED** [review]
Good stuff

File: deploy.sh [hunk 3]
```diff
@@ -12,11 +14,11 @@ minikube start --vm-driver=virtualbox
...
 kubectl create secret docker-registry regcred --docker-server=$REG_URL --docker-username=$REG_USERNAME --docker-password=$REG_PASSWORD --docker-email=$REG_EMAIL
-kubectl create configmap match-db-config --from-file match-db/init.sql -o=yaml
-kubectl create configmap user-db-config --from-file user-db/init.sql -o=yaml
-kubectl create configmap auth-db-config --from-file auth-db/init.sql -o=yaml
+kubectl create configmap match-db-config --from-file $BASEDIR/match-db/init.sql -o=yaml
+kubectl create configmap user-db-config --from-file $BASEDIR/user-db/init.sql -o=yaml
+kubectl create configmap auth-db-config --from-file $BASEDIR/auth-db/init.sql -o=yaml
 
-for dir in kube/*
+for dir in $BASEDIR/kube/*
```
Comment (c-harding) [inline deploy.sh]:
```suggestion
for dir in "$BASEDIR/kube/"*
```

**Review by c-harding — CHANGES_REQUESTED** [review]
This doesn’t work in bash when there’s a space in the path

**Comment by jaylees14** [comment]
> This doesn’t work in bash when there’s a space in the path

Nice catch. Although anyone with a space in a directory name deserves to be punished

**Review by c-harding — APPROVED** [review]

**Pull request merged**